tests/test_diagnostics.py

```diff
@@ -8,25 +8,35 @@
 
 
 @pytest.fixture(name='simple_field')
-def field_fixt():
+def field_fixt(tmp_path):
     """Pytest fixture for FieldDiagnostic class"""
     sim_dic = {"Grid": {"N": 2, "r_min": 0, "r_max": 1},
                "Clock": {"start_time": 0,
                          "end_time": 10,
                          "num_steps": 100},
                "Tools": {"ExampleTool": {}},
-               "PhysicsModules": {"ExampleModule": {}}}
-    field_dic = {"component": "Component",
-                 "field": "Field",
-                 "output_type": "csv",
-                 "filename": "output.csv",
-                 "dump_interval": 1}
-    field = FieldDiagnostic(Simulation(sim_dic), field_dic)
+               "PhysicsModules": {"ExampleModule": {}},
+               "Diagnostics": {
+                   "directory": f"{tmp_path}/default_output",
+                   # "clock": {},
+                   "field": [
+                       {"component": "Component",
+                        "field": "Field",
+                        "output_type": "csv",
+                        "filename": "output.csv",
+                        "dump_interval": 1}
+                       ]
+               }
+               }
+    sim = Simulation(sim_dic)
+    sim.read_diagnostics_from_input()
+    field = sim.diagnostics[0]
     field._owner.read_clock_from_input()
     field._owner.read_grid_from_input()

```

Comment:
This PR is looking really good!

Just one final (minor) suggestion:
```suggestion
    sim.read_diagnostics_from_input()
    sim.read_clock_from_input()
    sim.read_grid_from_input()
    field = sim.diagnostics[0]
```